node/node.ino

```diff
@@ -270,12 +269,12 @@ void sleep(uint32_t sleep_time) {
         delay(sleep_time);
     } else {
         // Other nodes sleep for real
-        int sleep_time_left = (int)sleep_time;
         uint32_t time_before = correct_millis();
 
-        while (sleep_time_left > 0) {
-            sleep_time_left -= Watchdog.sleep(sleep_time_left);
+        while (sleep_time > INT16_MAX) {
```

Comment:
sleep_time is uin32_t, so it's *unsigned*. Inside the while loop, the value returned by Watchdog.sleep() is subtracted from sleep_time. Watchdog.sleep() can return a value greater than the sleep time it is given(look in the watchdog code). This means that sleep_time can overflow into 4 bil and this while loop will run (almost) forever.

Reply:
the value returned by Watchdog.sleep() is int, so it's signed. Can we cast the value returned to unsigned? This should limit it to 16 bit (I'm assuming it only returns positive values)

Reply:
The problem is not with signed/unsigned. If the "while (sleep_time > INT16_MAX)" is entered and sleep_time_left is less than 15 at the last loop, then Watchdog.sleep will return 15(it never sleeps less than 15), so sleep_time_left will go negative, which in unsigned terms means it will go huge.

Reply:
Nevermind everything above. It won't happen because while loop ends when sleep_time < INT16_MAX.

*But* this is another bug, the while loop ends too early and then there's only one sleep. If sleep_time is INT16_MAX  - 1, then there's only one sleep which is 8 seconds max, so 50s or so will be missing.